src/main/java/com/seailz/discordjar/model/guild/Guild.java

```diff
@@ -1152,6 +1156,503 @@ public StringFormatter formatter() {
+                commaDelimitedSnowflakesString.append(r.id());
+                commaDelimitedSnowflakesString.append(",");
+            }
+        }
+        commaDelimitedSnowflakesString.append("]");
+
+        JSONArray snowflakes = new JSONArray();
+        roles.forEach(r -> snowflakes.put(r.id()));
+
+        System.out.println(new JSONObject()
+                .put("days", days)
+                .put("include_roles", commaDelimitedSnowflakesString).toString());
+        System.out.println(snowflakes);
+
+        try {
+            DiscordResponse req = new DiscordRequest(
+                    new JSONObject()
+                            .put("days", days)
+                            .put("include_roles", commaDelimitedSnowflakesString),
+                    new HashMap<>(),
+                    URLS.GET.GUILDS.PRUNE.replace("{guild.id}", id),
+                    discordJar,
+                    URLS.GET.GUILDS.PRUNE,
+                    RequestMethod.GET
+            ).invoke();
+            return req.body().getInt("pruned");
+        } catch (DiscordRequest.UnhandledDiscordAPIErrorException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Begins a prune operation on the guild. This requires the {@code KICK_MEMBERS} permission.
+     * @param days The amount of days to check users against, from 1-30.
+     */
+    public void prune(int days, List<Role> includedRoles) {
+        if (!Checker.inRange(1, 30, days, () -> Logger.getLogger("[DISCORD.JAR]").severe("Days cannot be outside 1-30!"))) return;
+
+        StringBuilder commaDelimitedSnowflakesString = new StringBuilder();
+        if (includedRoles.size() == 1) commaDelimitedSnowflakesString.append(includedRoles.get(0).id());
+        else {
+            for (Role r : includedRoles) {
+                commaDelimitedSnowflakesString.append(r.id());
+                commaDelimitedSnowflakesString.append(",");
+            }
+        }
+
+        try {
+            new DiscordRequest(
+                    new JSONObject()
+                            .put("days", days)
+                            .put("include_roles", commaDelimitedSnowflakesString),
+                    new HashMap<>(),
+                    URLS.POST.GUILDS.PRUNE.replace("{guild.id}", id),
+                    discordJar,
+                    URLS.POST.GUILDS.PRUNE,
+                    RequestMethod.POST
+            ).invoke();
+        } catch (DiscordRequest.UnhandledDiscordAPIErrorException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Returns a list of Guild {@link Member}s whose names (and optionally, nicknames) contain a filter string.
+     * @param filter The username filter string.
+     * @param includeNicknames Whether to include members whose nicknames contain the filter string.
+     * @return A list of Guild {@link Member}s
+     */
+    public List<Member> getMembersByName(String filter, boolean includeNicknames) {
```

Comment:
use https://discord.com/developers/docs/resources/guild#search-guild-members for this